src/Domain/Services/CreateAccount.php

```diff
@@ -19,26 +21,37 @@ class CreateAccount {
 	 *
 	 * @var WC_Email
 	 */
-	public $new_account_email = null;
+	public $new_account_email;
+
+	/**
+	 * Reference to the Package instance
+	 *
+	 * @var Package
+	 */
+	private $package;
 
 	/**
 	 * Constructor.
+	 *
+	 * @param Package $package An instance of (Woo Blocks) Package.
 	 */
-	public function __construct() {
+	public function __construct( Package $package ) {
+		$this->package = $package;
 	}
 
 	/**
-	 * Init - register custom hook handler.
+	 * Init - register handlers for WooCommerce core email hooks.
 	 */
 	public function init() {
-		// Register handler for signup during block-based checkout.
-		add_action( 'woocommerceblocks_create_account_for_order_request', [ $this, 'from_order_request' ], 10, 2 );
-
 		// Override core email handlers to add our new improved "new account" email.
 		add_action(
 			'woocommerce_email',
 			function ( $wc_emails_instance ) {
-				$this->new_account_email = require_once dirname( __DIR__ ) . '/class-wc-email-customer-new-account-blocks.php';
+				// Late instantiate CustomerNewAccount because it depends on WC_Email,
+				// which is not available at `init()` time.
+				if ( ! $this->new_account_email ) {
+					$this->new_account_email = new CustomerNewAccount( $this->package );
+				}
```

Comment:
Good point! Maybe this was more relevant in a previous iteration ... ? Made this change here: 89042da92decd0edbba6d836b79e7fa296987301

In some previous iteration, I hit `WC_Email undefined` errors because this was initialised too early, but I don't recall whether I actually needed it in a member 🤔